packages/ses/src/module-load.js

```diff
@@ -254,19 +242,24 @@ function* loadWithoutErrorAnnotation(
       return aliasRecord;
     }
 
-    throw TypeError('Unnexpected RedirectStaticModuleInterface record shape');
+    // A (legacy) behavior: If we do not recognize the module descriptor as a
+    // module descriptor, we assume that it is a module source (record):
```

Comment:
At some point, we’ll want to add options for strict portability and this case will become an exception in that mode. We would stop trying to differentiate module sources from module descriptors.

Reply:
I'm fine with the code here in this PR. But I don't understand the comment. Why not continue to differentiate module sources from module descriptors?

Reply:
We’ve discriminated all the module descriptors up to this point, but to preserve backward compatibility with import hooks that simply returned a module source (not a module source wrapped in a module descriptor), we have to fall through here.

We do have a clear path to deprecate this path, though, which I hope to get to separately.

Reply:
> which I hope to get to separately

Worth filing an issue?